src/core/services/interaction/interaction.js

```diff
@@ -0,0 +1,58 @@
+angular
+  .module('material.core.interaction', [])
+  .service('$mdInteraction', MdInteractionService);
+
+function MdInteractionService($timeout) {
+  var body = angular.element(document.body);
+  var _mouseEvent = window.MSPointerEvent ? 'MSPointerDown' : window.PointerEvent ? 'pointerdown' : 'mousedown';
+  var buffer = false;
+  var timer;
+  var lastInteractionType;
+  var inputMap = {
+    'keydown': 'keyboard',
+    'mousedown': 'mouse',
+    'mouseenter': 'mouse',
+    'touchstart': 'touch',
+    'pointerdown': 'pointer',
+    'MSPointerDown': 'pointer'
+  };
+  var pointerMap = {
+    2: 'touch',
```

Comment:
# Input Map
MSPointerdown < IE11: https://msdn.microsoft.com/de-de/library/windows/apps/hh465891.aspx
Pointerdown >= IE11: https://msdn.microsoft.com/de-de/library/windows/apps/hh465891.aspx
**Example** -> touchstart: Is associated with a touch interaction (so return the api value: "touch")
The rest you definitely  know :+1: 

#  Pointer Map:
https://msdn.microsoft.com/de-de/library/windows/apps/hh466130.aspx
MSPOINTER_TYPE_TOUCH: 0x00000002
MSPOINTER_TYPE_PEN: 0x00000003
MSPOINTER_TYPE_MOUSE: 0x00000004

Then I will validate the right pointer event type :)

I hope, know its clear :)
I'm up for more questions